src/interface/update.hpp

```diff
@@ -34,48 +34,50 @@ namespace parthenon {
 namespace Update {
 
 template <typename T>
-TaskStatus FluxDivergence(std::shared_ptr<T> &in, std::shared_ptr<T> &dudt_obj);
+TaskStatus FluxDivergence(T *in, T *dudt_obj);
 
 template <typename F, typename T>
-void UpdateData(const std::vector<F> &flags, T &in, T &dudt, const Real dt, T &out) {
-  const auto &in_pack = in->PackVariables(flags);
-  auto out_pack = out->PackVariables(flags);
-  const auto &dudt_pack = dudt->PackVariables(flags);
+TaskStatus SumData(const std::vector<F> &flags, T *in1, T *in2, const Real w1,
+                   const Real w2, T *out) {
```

Comment:
I think that in1 and in2 could be made const, though that would require adding const to the methods in the meshblock_data class and what ever other class is passed in. The PackVariables methods that I have seen are creating a copy and returning it so I don't really see why this can't be done.

Reply:
It's possible I'm misinterpreting something, but my belief is that the `PackVariables` calls don't work with a `const T *` because the caching modifies the internal state of the object.  So the `PackVariables` calls can't be marked `const`.  I _think_ you could use `T* const in1`, for example, since that just implies that pointer can't be changed, rather than implying that the object pointed to can't be changed.  I'm not sure marking it `T* const` is very helpful, though.

Reply:
Oh, I see what you mean, that's unfortunate, I skimmed over the helper methods in meshblock_data. Generally I think it is good to use const liberally even if it is only fixing the pointer address, but like you said in this instance the benefits are... minimal.